scripts/quests/sandoria/PLD_AF2_A_Boys_Dream.lua

```diff
@@ -0,0 +1,235 @@
+
+local quest = Quest:new(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.A_BOY_S_DREAM)
+
+quest.reward =
+{
+    item     = xi.items.GALLANT_LEGGINGS,
+    fame     = 40,
+    fameArea = xi.quest.fame_area.SANDORIA,
+}
+
+quest.sections =
+{
+    -- Section: Quest available.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getMainJob() == xi.job.PLD and
+                player:getMainLvl() >= xi.settings.main.AF2_QUEST_LEVEL and
+                player:getQuestStatus(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.SHARPENING_THE_SWORD) == QUEST_COMPLETED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getVar("Quest[0][91]Prog") == 0 then
+                        return quest:progressEvent(41) -- Starts Quest.
+                    elseif player:getVar("Quest[0][91]Prog") == 1 then --declined first
+                        return quest:event(40)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [41] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    else
+                        quest:setVar(player, 'Prog', 1) -- Declined First offering.
+                    end
+                end,
+
+                [40] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 2 then
+                        return quest:event(46)
+                    elseif questProgress == 3 then
+                        return quest:event(39) -- During Quest "A Boy's Dream" (after exoroche cs)
+                    elseif questProgress == 4 then
+                        return quest:event(60) -- During Quest "A Boy's Dream" (after trading bug) madame ?
+                    elseif questProgress == 5 then
+                        return quest:progressEvent(47) -- During Quest "A Boy's Dream" (after trading odontotyrannus)
+                    elseif questProgress >= 6 then
+                        return quest:progressEvent(25) -- During Quest "A Boy's Dream" (after Zaldon CS)
```

Comment:
Is it intended to have this dialogue continuously play out? It sets prog to 7, but checks for >= 6. Which would give the player this dialogue indefinitely. 
But you're using progressEvent which makes it seem like it shouldn't be the case.